### PR TITLE
Skeleton Tool: Use dashed outline for parent-child relationships

### PR DESCRIPTION
**Pull request by Aa20475**
This is a PR for issue #1588

### CODE REVIEW

**Comment by morevnaproject** [comment]
@Aa20475 Please fix LGTM warning.

File: synfig-studio/src/gui/duckmatic.cpp [hunk 1]
```diff
@@ -2968,7 +2968,9 @@ Duckmatic::add_to_ducks(const synfigapp::ValueDesc& value_desc,etl::handle<Canva
             }
         }
 
-        bone_transform_stack.push(new Transform_Rotate(guid, bone.get_angle()));
+        etl::handle<Bezier> bezier = new Bezier();
+
+    	bone_transform_stack.push(new Transform_Rotate(guid, bone.get_angle()));
```
Comment (morevnaproject) [inline synfig-studio/src/gui/duckmatic.cpp:2973]: Indentation.

File: synfig-studio/src/gui/duckmatic.cpp [hunk 2]
```diff
@@ -2992,8 +2994,50 @@ Duckmatic::add_to_ducks(const synfigapp::ValueDesc& value_desc,etl::handle<Canva
                                                                      0.0f),             // location
                                                           value_desc));                 // value_desc
             add_duck(duck);
+            bezier->p1 = bezier->c1 = duck;
         }
 
+        //parent tip fake
```
Comment (morevnaproject) [inline synfig-studio/src/gui/duckmatic.cpp:3000]: Indentation

File: synfig-studio/src/gui/duckmatic.cpp [hunk 3]
```diff
@@ -2992,8 +2994,50 @@ Duckmatic::add_to_ducks(const synfigapp::ValueDesc& value_desc,etl::handle<Canva
...
+				    origin = m.get_transformed(origin);
+
+				    //Obtaining Tip's location in global coordinates
+					origin += Point(bscale*bm.axis(0)[0],bscale*bm.axis(0)[1]);
+
+					//Converting tip's coordinates into local bone coordinates 
+				    m = bone.get_animated_matrix().get_inverted();
+				    origin = m.get_transformed(origin);
+
+				    synfigapp::ValueDesc value_desc(bone_value_node, bone_value_node->get_link_index_from_name(recursive ? "scalex" : "scalelx"), orig_value_desc);
```
Comment (morevnaproject) [inline synfig-studio/src/gui/duckmatic.cpp:3026]: This line introduces LGTM warning - https://lgtm.com/projects/g/synfig/synfig/rev/pr-0895b82cba7353ae5ed4fae8e19ccf6a405b652f
Please fix.

File: synfig-studio/src/gui/duckmatic.cpp [hunk 4]
```diff
@@ -2992,8 +2994,50 @@ Duckmatic::add_to_ducks(const synfigapp::ValueDesc& value_desc,etl::handle<Canva
...
+				    parent_bone = (*parent_bone.get_parent())(time).get(Bone());
+				    Matrix m = parent_bone.get_animated_matrix();
+				    Real scale = parent_bone.get_scalelx();
+				    origin[0]*=scale;
+
+				    //Changing parent's origin to global coordinates
+				    origin = m.get_transformed(origin);
+
+				    //Obtaining Tip's location in global coordinates
+					origin += Point(bscale*bm.axis(0)[0],bscale*bm.axis(0)[1]);
```
Comment (morevnaproject) [inline synfig-studio/src/gui/duckmatic.cpp:3020]: Indentation

**Comment by Aa20475** [comment]
Alright!

File: synfig-studio/src/gui/duckmatic.cpp [hunk 5]
```diff
@@ -3128,7 +3170,7 @@ Duckmatic::add_to_ducks(const synfigapp::ValueDesc& value_desc,etl::handle<Canva
             duck->set_origin(tip_duck);
             add_duck(duck);
         }
-
+
```
Comment (rodolforg) [inline synfig-studio/src/gui/duckmatic.cpp:3173]: indentation

**Comment by morevnaproject** [comment]
@Aa20475 Do you need any help on how to fix LGTM warning?

**Comment by Aa20475** [comment]
I fixed it?

**Comment by morevnaproject** [comment]
Ah, right, it was a part of "Indentation fixes" commit (sigh).

**Comment by Aa20475** [comment]
Yes!

**Pull request merged**